### PR TITLE
Add weights and axis parameters to ``vonmisesmle``

### PR DESCRIPTION
**Pull request by FelipeCybis**
### Add weights and axis parameters to ``vonmisesmle``

This pull request is to address the lack of "weights" and "axis" parameters on the ``circstats`` function ``vonmisesmle``.
The "axis" parameter that existed but was not actually being used allows for fast usage of 2D numpy arrays.
The "weights" parameter is very convenient since it is very common to have data for binned angles.

The appropriate tests were added to `` test_circstats.test_vonmisesmle`` in [4c6172f](https://github.com/astropy/astropy/pull/14533/commits/4c6172fe5a366aa6987fcab2a9eb8d02150f5e97), but here is a minimum example using scipy's vonmises PDF:

```
>>> from astropy import stats
>>> import scipy
>>> x = np.linspace(0,2*np.pi,100)
>>> mu, kappa = stats.circstats.vonmisesmle(x, weights=scipy.stats.vonmises.pdf(x, loc = np.pi/2, kappa=4))
>>> mu, kappa
(1.56976133088732, 3.9514188541707314)
```

Or for 2D arrays:

```
>>> x = np.linspace(0,2*np.pi,100)
>>> xx = np.tile(x, [4,1])*u.rad
>>> vm = np.stack([
    scipy.stats.vonmises.pdf(x, loc = np.pi/2, kappa=4),
    scipy.stats.vonmises.pdf(x, loc = np.pi/2, kappa=1),
    scipy.stats.vonmises.pdf(x, loc = np.pi/4, kappa=4),
    scipy.stats.vonmises.pdf(x, loc = np.pi/8, kappa=8)])
>>> stats.circstats.vonmisesmle(xx, weights=vm, axis=1)
(<Quantity [1.56976133, 1.55292537, 0.77316812, 0.37759809] rad>,
 <Quantity [3.95141885, 0.98695686, 3.91289117, 7.95905254]>)
```

Fixes # There is not opened issue, since it is an enhancement.

### CODE REVIEW

File: docs/changes/circstats/14533.feature.rst
Comment (pllim) [inline docs/changes/circstats/14533.feature.rst:1]: Please don't create your own category. This should go under https://github.com/astropy/astropy/tree/main/docs/changes/stats . You also need to squash out the commits after you have undone this new dir creation. Thanks!

**Comment by pllim** [comment]
Thanks, you might also be interested in the discussions at #13749

File: docs/changes/circstats/14533.feature.rst
Comment (FelipeCybis) [reply docs/changes/circstats/14533.feature.rst:1]: Ah, sorry, I thought each folder had the name of a module but no one had done a PR to circstats before and that's why there was no folder for it... Will do that!

Comment (FelipeCybis) [reply docs/changes/circstats/14533.feature.rst:1]: Done :)

File: docs/changes/stats/14533.feature.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+This pull request is to address the lack of "weights" and "axis" parameters on the circstats function ``vonmisesmle``.
```
Comment (pllim) [inline docs/changes/stats/14533.feature.rst:1]: I would rephrase the change log text to the tone of what you want user to know about the changes when they read the release change log. They would not care it was from a pull request. Feel free to look at existing change log texts for examples. Thanks!

**Comment by FelipeCybis** [comment]
Thanks for directing me to https://github.com/astropy/astropy/issues/13749. Indeed, could be interesting taking the best of both packages in the long run. Particularly I could certainly not work without weighing my arrays (hence the PR ;)). Also if you guys decide on adapting the kappa estimation using the scipy method it should be straight forward with the weights and axis params.

Also, tell me if I can still improve this PR in some way :)

**Comment by pllim** [comment]
@FelipeCybis , sorry for the delay. Coverage is still complaining but the log is too old now, so I cannot see if the coverage upload failed or is it because of something else. Could you please rebase? 

@larrybradley , are you able to review this before feature freeze?

Thanks!

File: astropy/stats/circstats.py [hunk 2]
```diff
@@ -494,15 +494,24 @@ def vtest(data, mu=0.0, axis=None, weights=None):
...
+    kappa1 = np.where(
+        np.logical_and(0 <= x, x < 0.53),
+        2.0 * x + x * x * x + (5.0 * x**5) / 6.0,
+        0)
+    kappa2 = np.where(
+        np.logical_and(0.53 <= x, x < 0.85),
+        -0.4 + 1.39 * x + 0.43 / (1.0 - x),
+        0)
+    kappa3 = np.where(
+        np.logical_or(x < 0, 0.85 < x),
```
Comment (larrybradley) [inline astropy/stats/circstats.py:507]:
```suggestion
        np.logical_or(x < 0, 0.85 <= x),
```

**Review by larrybradley — CHANGES_REQUESTED** [review]
Thanks, @FelipeCybis.  LGTM, with one minor requested change.

**Comment by pllim** [comment]
pre-commit.ci autofix

File: astropy/stats/tests/test_circstats.py [hunk 3]
```diff
@@ -146,3 +146,62 @@ def test_vonmisesmle():
...
+    )
+    assert_allclose(
+        answer[1], vonmisesmle(data_to_weigh, weights=weights)[1], atol=1e-5
+    )
+
+    # testing for axis argument (stacking the data from the first test)
+    data = np.array(
+        [
+            [
+                3.3699057,
```
Comment (pllim) [inline astropy/stats/tests/test_circstats.py:173]: Seems like `black` overdid it a little but this is what would make the precommit bot happy. 🤷

Comment (larrybradley) [reply astropy/stats/tests/test_circstats.py:173]: I really don't like this formatting. We could manually reformat and turn off black (`fmt: off`), but let's just get this in before the freeze.

**Review by larrybradley — APPROVED** [review]
Thanks, @FelipeCybis

**Comment by pllim** [comment]
Thanks, all!
